### PR TITLE
Adapter v4.0.0 Release Notes

### PR DESCRIPTION
**Pull request by OsirisTerje**

### CODE REVIEW

File: docs/articles/vs-test-adapter/Adapter-Release-Notes.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-Release-Notes.md:7]: Unclear whether you mean there are plans to backport support into prior versions, or whether the intention is that folks will have to upgrade to 4.x in order to use it with Visual Studio 2022.

File: docs/articles/vs-test-adapter/Known-Problems.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Known-Problems.md:7]:
```suggestion
    1  Adapter versions in the 4.XX series support the `explicit` keyword in all Visual Studio versions.
```

File: docs/articles/vs-test-adapter/Known-Problems.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Known-Problems.md:9]:
```suggestion
    2 Adapter versions in the 3.XX series:  From version 16.2 (Visual Studio 2019) the `Explicit` keyword is no longer explicitly supported by Visual Studio.  If you want to use it, it does exist as a Category, so you can use it as a category filter.
```

File: docs/articles/vs-test-adapter/Tips-And-Tricks.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Tips-And-Tricks.md:190]:
```suggestion
When set to `true`, the default, the name of the test is added as a prefix in front of each console line output.  This only applies to the output in the output window.
```

**Review by SeanKilleen — COMMENTED** [review]
Minor suggestions for you to accept if you like.

Also FYI, I made a commit to remove two additional blank lines which allows the markdown linter to pass. 👍

File: docs/articles/vs-test-adapter/Adapter-Release-Notes.md
Comment (OsirisTerje) [reply docs/articles/vs-test-adapter/Adapter-Release-Notes.md]: Updated the paragraph to make it more clear.  IF the VS team can manage to fix whatever is causing this in the versions they release, we will be fine.  Anything we do will result in an upgrade, and then the 4.X versions are that upgrade.

File: docs/articles/vs-test-adapter/Adapter-Installation.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-Installation.md:10]:
```suggestion
* Use the Extension Manager  (Deprecated and pending future removal; see notes below.)
```

File: docs/articles/vs-test-adapter/Adapter-Installation.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-Installation.md:26]:
```suggestion
> [!WARNING]
> The use of the VSIX extension manager is deprecated.  It still works, but at some time support for it will be removed from Visual Studio, and also newer versions of the adapter. We recommend you try to move to nuget based adapters.
```

I'm thinking this will give it some flair.

File: docs/articles/vs-test-adapter/Adapter-Installation.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-Installation.md:28]:
```suggestion
```

File: docs/articles/vs-test-adapter/Adapter-Installation.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-Installation.md:41]:
```suggestion
The use of the VSIX extension manager is deprecated.  It still works, but at some time support for it will be removed from Visual Studio, and also newer versions of the adapter.
```

File: docs/articles/vs-test-adapter/Adapter-Installation.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-Installation.md:49]:
```suggestion
The Extension will be installed to Visual Studio itself, and will work for all projects you use. All users of your solution need to install the extension. If you use TFS Build, you must also install the extension to the build system there.
```

File: docs/articles/vs-test-adapter/Adapter-License.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-License.md:3]: Can you please link to a discussion or agreement somewhere to change the licensing and add you to the copyright? Just want to make sure we're doing this by the book.

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:3]:
```suggestion
Trace and Debug output is by default not sent to the console output, or anywhere else.  If you want to have this output, you must add your own `TraceListener`.
```

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:5]:
```suggestion
## Implementing the `TraceListener` in your code
```

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:11]:
```suggestion
```csharp
```

Though both are fine, the docs are standardized on `csharp`.

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:29]:
```suggestion
If you place the code above outside of any namespace it will apply to the whole assembly.
```

Removing the backticks here as it does not refer to code or an explicit choice as far as I can tell.

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:31]:
```suggestion
If you'd like to only apply to a given test fixture/class, you can add the `StartTest and EndTest` methods to the given test fixture/class.
```

Trying to remove "just" style words from the docs.

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:35]:
```suggestion
Given the following verbose test:
```

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:37]:
```suggestion
```csharp
```

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:52]:
```suggestion
With a `TraceListener`, the output in Visual Studio Test Explorer is:
```

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:54]:
```suggestion
![Trace Debug output showing the lines outputted by the TraceListener](../../images/TraceDebug1.png)
```
Adding a little more to the caption

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:56]:
```suggestion
Without the `TraceListener`, the blocks in red will not be present.
```

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:58]:
```suggestion
Running `dotnet test -v n`, the output will be:
```

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:60]:
```suggestion
![DotNet Trace Debug output showing the lines outputted by the TraceListener](../../images/TraceDebug2.png)
```

File: docs/articles/vs-test-adapter/Trace-and-Debug.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Trace-and-Debug.md:64]:
```suggestion
This issue have been discussed at [Issue 718](https://github.com/nunit/nunit3-vs-adapter/issues/718) and [Issue 301](https://github.com/nunit/nunit3-vs-adapter/issues/301). Use of `TraceListener` suggested by [Frans Bouma](https://github.com/FransBouma).
```

I like the way you give a shout-out to Frans here too. 👍

File: docs/articles/vs-test-adapter/Adapter-Installation.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-Installation.md:12]:
```suggestion
```

Suggesting removal of extra line per markdown linting.

File: docs/articles/vs-test-adapter/Adapter-Installation.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-Installation.md:13]:
```suggestion
```

Suggesting removal of extra line per markdown linting.

File: docs/articles/vs-test-adapter/Adapter-License.md
Comment (SeanKilleen) [inline docs/articles/vs-test-adapter/Adapter-License.md:23]:
```suggestion
THE SOFTWARE.

```

File should end with a newline, per markdown linting.

**Review by SeanKilleen — COMMENTED** [review]
A bunch of minor comments in the form of PR suggestions. Feel free to accept those you agree with and reject those you don't, or to start a conversation under any of them. 👍

**Comment by OsirisTerje** [comment]
Thanks a ton, @SeanKilleen , for all your comments and suggestions.  I really appreciate it :-)

**Comment by SeanKilleen** [comment]
No problem! Happy to help.

This looks ready to go, but I'll mark it as "pending changes" so that we can get the question about the license change answered. Once that's answered we'll be good to merge this. Thanks!

**Review by SeanKilleen — CHANGES_REQUESTED** [review]
Marking as "request changes" since we'll need to have the traceability on the license changes. Should be able to quickly resolve. 👍

File: docs/articles/vs-test-adapter/Adapter-License.md
Comment (OsirisTerje) [reply docs/articles/vs-test-adapter/Adapter-License.md:3]: I just noticed this recently, and it is probably an oversight from years back.  @CharliePoole can confirm.  The copyright for the adapter is by Charlie Poole 2011 and me 2014, and the license is MIT.  This can be seen in the NUnit3Adapter repo.

Comment (CharliePoole) [reply docs/articles/vs-test-adapter/Adapter-License.md:3]: @SeanKilleen The documentation copy of the license is secondary. If it's provided, it must always be a copy of the LICENSE file provided with the software, which is where the history lives. My guess is that some old file had not been updated at the point where you converted the docs to docfx.

I modified the license to include Terje as a co-owner of the software in 2017. See https://github.com/nunit/nunit3-vs-adapter/commit/21624d1348d2274e094914be5e98ef312cd3523b

@OsirisTerje You should not be updating my dates in the header after I stopped contributing. You could also reasonably decide that 4.0 is now a new work of your creation and remove my name. That's what I did on the release of NUnit 3 (framework, console and GUI) after verifying that virtually no remaining code had been contributed by others. I also added a notice file that mentioned the original creators - not required by MIT but it seemed only polite. It's a question of your judgement in my opinion.

In case anyone should wonder, Terje and I are the sole copyright holders for the adapters. The Core Team has no involvement in the decision, although an individual member could challenge the copyright if they felt it was wrong.

**Pull request merged**
